h/accounts/views.py

```diff
@@ -467,12 +467,10 @@ def disable_user(self):
         if err is not None:
             return err
 
-        username = appstruct['username']
-        pwd = appstruct['pwd']
+        user = User.get_by_userid(
+            self.request.domain, self.request.authenticated_userid)
 
-        # Password check
-        user = User.get_user(username, pwd)
-        if user:
+        if User.validate_user(user, appstruct['pwd']):  # Password check.
```

Comment:
Actually this password check is unnecessary now, isn't it? It looks like the disable user form once allowed users to disable any user account that they know the password of? But now, it only lets you disable your own account which you're already logged-in to